container/falcon-container-terraform/vm-admin.ds.sh.tpl

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+
+export HOME=/root
+
+main(){
+    export DEBIAN_FRONTEND=noninteractive
+    install_deps
+    fetch_falcon_secrets
+    download_build_falcon_sensor
+    push_falcon_sensor_to_image_registry
+    deploy_falcon_helm
+    deploy_vulnerable_app
+    wait_for_vulnerable_app
+}
+
+install_gofalcon() {
+    gofalcon_version=0.2.2
+    pkg=gofalcon-$gofalcon_version-1.x86_64.deb
+    wget -q -O $pkg https://github.com/CrowdStrike/gofalcon/releases/download/v$gofalcon_version/$pkg
+    apt install ./$pkg > /dev/null
+}
+
+install_deps(){
+    sudo apt-get update
+    sudo curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+    sudo apt install docker.io -y && sudo systemctl enable docker --now
+    sudo az login --identity
+    sudo az aks get-credentials --name "${AKS_CLUSTER_NAME}" --resource-group "${RG_NAME}" --admin
+    sudo az aks install-cli
+    install_gofalcon
+    sudo mkdir -p /yaml
+    wget -q -O /yaml/vulnerable.example.yaml https://raw.githubusercontent.com/isimluk/vulnapp/master/vulnerable.example.yaml
+}
+
+download_build_falcon_sensor(){
+    tmpdir=$(mktemp -d)
+    pushd "$tmpdir" > /dev/null
+      git clone https://github.com/CrowdStrike/Dockerfiles.git
+      cd Dockerfiles
+      falcon_sensor_download --os-name=Ubuntu --os-version=14/16/18/20 --sensor-version=latest
+      local_repo=falcon-node-sensor
+      tag=ubuntu
+      local_tag="$local_repo:$tag"
+      docker build --no-cache --build-arg \
+        BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ') \
+        --build-arg VCS_REF=$(git rev-parse --short HEAD) \
+        --build-arg FALCON_PKG=$(ls falcon-sensor_6*) \
```

Comment:
This will break when falcon-sensor_7 is released.

I think, I should modify falcon_sensor_download to just output the filename of downloaded sensor. To enable us to grab the downloaded sensor from the tool output to a shell variable.